validator_test.go

```diff
@@ -973,7 +985,10 @@ func TestValidationFailOnDefaults(t *testing.T) {
 	for i, test := range tests {
 		t.Run(fmt.Sprintf("Test config (%v): %#v", i, test), func(t *testing.T) {
 			err := c.Unpack(test)
-			assert.True(t, err != nil)
+			if err == nil {
+				t.Fatalf("test:%v error:%v", spew.Sdump(test), err)
+			}
```

Comment:
We have this pattern in 3 places now, maybe more? Consider to provide a helper function like this (also guarntees that the message is and keeps unified between tests in the future):

```
func mustUnpack(t *testing.T, cfg *Config, to interface{}) {
   if err := c.Unpack(to); err != nil {
       ...
   }
}
```

Reply:
I'm keeping that in mind, for now I've removed some redundancy with the `validateErrorTemplate` variable.

Reply:
This makes the C programmer in me sad. Fortunately Printf is more secure in go. :)

For Printf style function I'd prefer to not use `const` or `vars` for 'templates'. Having the format string always with the Printf functions helps linters (e.g. go vet) to detect errors early. This is why I'm prefering a function here (can also be a function for logging/failing only), as this will also fix arguments we accept and force developers to follow up if they do any changes to the format string.

Reply:
I did this in two functions, `mustUnpack()` and `mustFailUnpack()`.